Xamarin.Forms.Core/FormattedString.cs

```diff
@@ -4,27 +4,31 @@
 using System.Collections.Specialized;
 using System.ComponentModel;
 using System.Linq;
-using System.Runtime.CompilerServices;
 
 namespace Xamarin.Forms
 {
 	[ContentProperty("Spans")]
-	public class FormattedString : INotifyPropertyChanged
+	public class FormattedString : BindableObject
 	{
 		readonly SpanCollection _spans = new SpanCollection();
 
 		public FormattedString()
 		{
-			_spans.CollectionChanged += OnCollectionChanged;
+			_spans.CollectionChanged += OnCollectionChanged;			
+			BindingContextChanged += FormattedString_BindingContextChanged;
+		}
+
+		private void FormattedString_BindingContextChanged(object sender, EventArgs e)
+		{
+			foreach (var span in Spans)
+				SetInheritedBindingContext(span, BindingContext);
```

Comment:
Sets Inherited BindingContext, from FormattedString all the way down to the Spans